ui/apps/platform/src/services/CollectionsService.ts

```diff
@@ -0,0 +1,234 @@
+
+export type GetCollectionSelectorsResponse = {
+    selectors: SelectorField[];
+};
+
+/**
+ * Create a new collection
+ *
+ * @param collection
+ *      The collection object details to be created
+ * @returns
+ *      The created collection object, with ID
+ */
+export function createCollection(
+    collection: CollectionRequest
+): CancellableRequest<CollectionResponse> {
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .post<CollectionResponse>(collectionsBaseUrl, collection, { signal })
+            .then((response) => response.data)
+    );
+}
+
+/**
+ * Updates an existing collection object.
+ *
+ * @param id
+ *      The ID of the collection to update
+ * @param collection
+ *      The new collection object details.
+ * @returns
+ *      The updated collection object
+ *
+ */
+export function updateCollection(
+    id: string,
+    collection: CollectionRequest
+): CancellableRequest<CollectionResponse> {
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .post<CollectionResponse>(`${collectionsBaseUrl}/${id}`, collection, { signal })
+            .then((response) => response.data)
+    );
+}
+
+/**
+ * Deletes a collection
+ *
+ * @param id
+ *      The ID of the collection to delete
+ */
+export function deleteCollection(id: string): CancellableRequest<Empty> {
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .delete<Empty>(`${collectionsBaseUrl}/${id}`, { signal })
+            .then((response) => response.data)
+    );
+}
+
+export type CollectionDryRunRequest = CollectionRequest & {
+    options: {
+        pagination: Pagination;
+        skipDeploymentMatching: boolean;
+    };
+};
+
+export type CollectionDryRunResponse = {
+    deployments: ListDeployment[];
+};
+
+/**
+ * Fetches the currently matching deployments for a collection based on the applied resource
+ * selectors and embedded collections. Note that the deployments in a collection are resolved
+ * dynamically and may change over time as new deployments are added to a cluster.
+ *
+ * @param dryRunRequest.resourceSelectors
+ *      The resource selector rules used to match deployments
+ * @param dryRunRequest.embeddedCollectionIds
+ *      An array of collection ids whose matching deployments should
+ *      be added to the result set.
+ * @param dryRunRequest.options.pagination
+ *      Pagination options for the dry run deployment results
+ * @param dryRunRequest.options.skipDeploymentMatching
+ *      This flag will skip the resolution of matching deployments on the back end
+ *      in order to do more efficient error checking. Used in order to determine if
+ *      a config is valid, without returning a full data payload.
+ *
+ * @returns A list of deployments that are resolved by the collection.
+ */
+export function dryRunCollection(
+    dryRunRequest: CollectionDryRunRequest
+    // TODO `ListDeployment` will make this impossible to paginate without loading the entire
```

Comment:
In order to implement the "Selector results" sidebar as designed in the mocks we need to aggregate all clusters and namespaces that match a deployment _name_, under that deployment name. The standard `ListDeployment` type will return individual deployments with a single cluster and namespace, even if there is a matching deployment name in another cluster.

Essentially we need something with a structure similar to the results of this GQL query:
```gql
query q {
  deployments(query: "Deployment:kube") {
    name
    namespaceObject {
      metadata { name }
      cluster { name }
    }
  }
}
```
but with the ability to accept `ResourceSelectors` to filter the results.

The BE team is currently looking to see if this is feasible to do over the `dryRunCollections` REST API.